README.md

```diff
@@ -2,24 +2,13 @@
 
 [![Join the chat at https://gitter.im/yelouafi/redux-saga](https://badges.gitter.im/yelouafi/redux-saga.svg)](https://gitter.im/yelouafi/redux-saga?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) [![npm version](https://img.shields.io/npm/v/redux-saga.svg?style=flat-square)](https://www.npmjs.com/package/redux-saga)
 
-An alternative Side Effects middleware (aka Asynchronous Actions) for Redux applications. Instead of dispatching Thunks which get handled by the `redux-thunk` middleware, you create *Sagas* to gather all your Side Effects logic in a central place.
+`redux-saga` is a library that aims to make asynchronous flows (e.g. data fetching) in React.js applications easier and better.
 
-This means application logic lives in 2 places:
+The mental model is that a saga is like a separate thread in your application that's solely responsible for asynchronousity. This thread can be started, paused and cancelled from the main application with normal redux actions, it has access to the full redux application state and it can dispatch redux actions as well.
```

Comment:
While this is correct. There are also things that are synchronous but not pure (like saving to a browser storage, clearing a timeout ID, ...). If we define Side Effects in the first paragraph, we can simply put here that a Saga is solely responsible for Side Effects, or to reflect the opinions of the library: redux-saga proposes to 'isolate' all sorts of Side Effects (including async calls) inside Sagas.